assets/shell-integration/wezterm.sh

```diff
@@ -486,9 +486,11 @@ __wezterm_semantic_precmd() {
 }
 
 function __wezterm_semantic_preexec() {
-  # Restore the original PS1/PS2
-  PS1="$__wezterm_save_ps1"
-  PS2="$__wezterm_save_ps2"
+  # Restore the original PS1/PS2 if set
+  if [ -n "$__wezterm_save_ps1" ]; then
+	  PS1="$__wezterm_save_ps1"
+	  PS2="$__wezterm_save_ps2"
+  fi
```

Comment:
should this block also unset `__wezterm_save_ps1`?

It seems to me that after running one command, there will always be something in `__wezterm_save_ps1`, so this feels like a one-time condition as written right now.

Reply:
I don't think PS1 changes all that much, so it's probably fine like this?
But if we don't unset it it's possible that we'd restore an old PS1, so I guess that it'd make more sense to unset it strictly speaking.

I don't remember in which exact conditions I had produced that but now you're pointing this out it might always have been in the first command after ssh'ing in and using ^C, which would make sense given after the first command the saved variable would always be set.
(It was fairly rare and took me a while to understand where that might have gotten wrong, I hadn't identified a pattern)

Anyway, I don't have strong opinions here, happy to add unset if you prefer

Reply:
I think it would be more consistent and obviously worthwhile if this unset it!

Reply:
Thanks, updated!